### ISSUE TITLE
Icon selection

### ISSUE THREAD

**Issue by NoNamePaul** [brittyazel/Neuron#100]
I have 2 little issues with the icon selection screen:
1. The searchbar sometimes doesn't appear. I have to click custom icon and cancel to make it reappear (Probably gone with the new UI together with the X button being invisible)
2. The search function appears to not function correctly at all:
    I tried search for the Cyclone Icon (Druid) in 3 different ways:
     - Typing "cyclone"
     - Typing the icon ID (136022 from wowhead)
     - Typing the icon's name "Spell_nature_earthbind" (from wowhead)
  None of these found the icon, just random other stuff or nothing at all.

It's really frustrating to go through all the icons to make the perfect interface.

**Comment by brittyazel** [comment]
Yeah I agree 100% that the search bar is broken. I thought I had the search bar hidden, but I guess you figured out how to get it to come back ;-)

The reason that the search is broken is because we can't search for icons based on Strings anymore, i.e. "cyclone". As of Legion, the only way to query an icon is to know the exact path location or the texture id, i.e. 445050504. Hence, since we're using the texture id's to populate the list we have lost our ability to have any "searchable". This was one of the very first bugs I encountered after taking over the project, so maybe I'll give it another go now that I'm more seasoned.

**Comment by NoNamePaul** [comment]
Maybe you could just give us an input field for an ID? We can easily search for it on wowhead:
So [cyclone](http://www.wowhead.com/icon=136022/spell-nature-earthbind) would be 136022?
Or is the texture id not the one shown in the wowhead url?

**Comment by brittyazel** [comment]
Hah, wouldn't that be nice. Nope what you listed is the SpellID, which is different from the texture ID. Fun right?!

I'm sure there's an elegant way to address this, I just have to be really clever, which only happens every once and a while lol
